CONTRIBUTING.md

```diff
@@ -1,16 +1,239 @@
+Describe your needs and use cases if any. Give detail on why this is important
+and/or necessary for RIOT.
+
+## Bug Reports
+[bug-reports]: #bug-reports
+
+While bugs are unfortunate, they're a reality in software. We can't fix what we
+don't know about, so please report liberally. If you're not sure if something
+is a bug or not, feel free to file a bug anyway.
+
+**If you believe reporting your bug publicly represents a security risk to
+RIOT users, please send an email describing the bug to security@riot-os.org**.
+
+Before reporting a bug, have a look at
+[opened bugs](https://github.com/RIOT-OS/RIOT/issues?q=state:open+type:issue+label:bug).
+Maybe someone has already reported your error.
+
+Once you have verified this, opening an issue is as easy as clicking on [this
+link](https://github.com/RIOT-OS/RIOT/issues/new?template=bug_report.md) and
+filling out the fields.
+
+Each bug report issue uses a template with 5 sections that are here to help
+maintainer understand your issue and eventually reproduce it:
+
+    #### Description
+
+    #### Steps to reproduce the issue
+
+    #### Expected results
+
+    #### Actual results
+
+    #### Versions
+
+To fill the `version` section, use the script provided in the RIOT git
+repository:
+```
+./dist/tools/ci/print_toolchain_versions.sh
+```
+
+In summary, try to put as much information as possible that you think could
+help maintainers or other developers to fix the bug quickly.
+
+## Pull Requests
+[pull-requests]: #pull-requests
+
+Pull Requests (PR) are the primary mechanism used to change RIOT. GitHub itself
+has some great documentation on
+[using the Pull Request feature][about-pull-requests].
+We use the [fork and pull model][development-models], where contributors push
+changes to their personal fork and create pull requests to bring those changes
+into the source repository.
+
+[about-pull-requests]: https://help.github.com/articles/about-pull-requests/
+[development-models]: https://help.github.com/articles/creating-a-pull-request-from-a-fork
+
+* [General rules](#general-rules)
+* [Git usage](#git-usage)
+
+### General rules
+[general-rules]: #general-rules
+
+* Before opening a new Pull Request, have a look at
+  [existing ones](https://github.com/RIOT-OS/RIOT/pulls). Maybe someone has
+  already opened one about the same thing and if it's the case, you might be
+  able to help with the contribution. Just comment on the PR and ask.
+
+* Each Pull Request form uses a template with 3 sections that are here to help
+  maintainer understand your contribution and help them testing it:
+
+      #### Contribution description
+
+      #### Testing procedure
+
+      #### Issues/PRs references
+
+  Please fill each section with as much information as possible.
+
+* Remember that smaller PRs tend to be merged faster, so keep your changes as
+  atomic as possible: don't hesitate to split your PRs into smaller ones.
```

Comment:
https://github.com/RIOT-OS/RIOT/pull/9527/files suggests a different wording for "atomic", which is a term which has been debated as being ill defined. The wording suggested in that PR is
"a manageable size.. should be confined to a single explainable change, and be runnable on its own."